drivers/adc/adc_stm32.c

```diff
@@ -592,7 +592,12 @@ static int start_read(const struct device *dev,
 	!defined(CONFIG_SOC_SERIES_STM32F1X) && \
 	!defined(STM32F3X_ADC_V2_5) && \
 	!defined(CONFIG_SOC_SERIES_STM32L1X)
+		LL_ADC_Disable(adc);
```

Comment:
Few lines above, the ADC has just been enabled after setting the oversampling bits, for the STM32L0x, but it seems that all soc requires ADSTART =0. 
Could we find and common combination to avoid disabling/waiting/enabling/disabling/re-enabling ?

Reply:
That is true for the G4 at least. Should we check the data sheets for all other STM32s? It seems like we can combine disabling the ADC for oversampling calibration setting.

![image](https://user-images.githubusercontent.com/10335942/145010750-5ba77425-51ea-4919-bd87-a05eb4243d97.png)

Reply:
@FRASTM in regards to your comment in the issue, in  RM0008, it recommends calibrating once after power-on, so I suppose that is why requesting calibration before starting an ADC read is not supported in the code. Do you concur?